### DISCUSSION TITLE
Provide clarity in documentation on "Use workflow from" Dropdown when running Manual Workflow

### DISCUSSION THREAD

**Discussion by colinbjohnson** [community/community#112371] [Actions]
### Select Topic Area

Product Feedback

#### Problem Description

When selecting the "Use workflow from" option in GitHub actions it is _not_ entirely clear what this option does. The reasons for this ambiguity are:

* The option "Use workflow from" implies that GitHub Actions will utilize a _workflow file_ from a given `ref` but does not make clear what `ref` will be utilized to select the _code_ to be utilized by the action itself
* No documentation exists that describes this option - doing a search on "[Use workflow from]"(https://github.com/search?q=org%3Adocs%20%22Use%20workflow%20from%22&type=code) returns no documents
* Through testing it is clear that this option selects the workflow file _and_ sets the ref of the code to be utilized

#### Proposed Resolution(s)

* it would be ideal if this option was documented in GitHub docs
* it would be ideal if selecting the "workflow file" and "code ref to be utilized during the action" was somehow clarified
  * this could be renaming the option to be `branch/tag/sha (aka: ref) to be utilized by workflow`

#### Proposed Documentation Addition

I wrote this up:

```
the `Use workflow from` option does the following:
- selects the “Workflow” file that will be utilized during workflow execution
- selects the code to be utilized during workflow execution
```

#### Addendum

I _wish_ I could provide advice to allow actions such as this (meaning selecting the code to be utilized on during a manual workflow execution) to be clear to any user, quickly. I tried my best with `branch/tag/sha (aka: ref) to be utilized by manual workflow`.

**Comment by jonw-cogapp** [comment]
My reading of this was _workflow file_. I had my doubts and so that's how I ended up here. The phrase "workflow execution context" would probably have made things clearer for me, but I could see an argument that this basically conveys the same possible meanings in a more complex way — but personally I think it points more towards the actual behavior.
